src/EbmlElement.cpp

```diff
@@ -601,21 +601,21 @@ filepos_t EbmlElement::OverwriteData(IOCallback & output, bool bKeepPosition)
 
   auto CurrentPosition = output.getFilePointer();
   output.setFilePointer(GetElementPosition() + HeaderSize);
-  auto Result = RenderData(output, true, bKeepPosition);
+  auto Result = RenderData(output, true, bKeepPosition ? WriteAll : WriteSkipDefault);
```

Comment:
I don't understand the intent behind this change. What does keeping the position have to do with the question whether or not to write the element?

Reply:
This is just the translation of the previous call into the new system. `bKeepPosition` would be used as the `bWithDefault` value. So if it's true it writes elements even if they are the default value (WriteAll), if it's false it only writes the elements if they are not the default value (SkipDefault).

The original logic may be flawed and it should be fixed separately. In fact when overwriting you'd probably want to tell which elements you want to write (the least possible).

This whole API is a recipe for broken files (which originally was a welcome possibility in libebml to be able to generate broken files to test against). It seems mkvtoolnix doesn't use it, there's only a test calling OverwriteHead(). So I'd rather remove the API altogether.

Reply:
Ah, I see. Thanks for the explanation. Yeah, MKVToolNix (or rather: `mkvpropedit`) only uses `OverwriteHead` for cases where it has to cover a one-byte space that's suddenly left by other operations; in such a case it forces the element size field to be one byte longer than necessary.

It never uses `OverwriteData`, though. If it has to overwrite a whole element, I think it simply uses `Render` & sets the file position manually.